### PR TITLE
fix valve altitude sync with geometry (1.2.9)

### LINKED ISSUE #187: valve geometry z - doesn't get updated in function of altitude
For the hydrants and all the elements that are edited through ``vw_element_*`` the altitude is synced with the geometry.

In the current model (1.2.7) the valves don't have this functionality although they had it when they were edited through the old ``vw_element_valve`` view.

Is this intentional?

### PR DESCRIPTION
**Pull request by 3nids**
should fix #187
@tudorbarascu can you check if it works for you?

### CODE REVIEW

**Comment by ponceta** [comment]
After pum final integration, do not forget to update travis line:

https://github.com/qwat/qwat-data-model/blob/master/.travis.yml#L44

File: update/delta/delta_1.2.9_fix_valve_alt_sync.sql [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	-- altitude is prioritary on Z value of the geometry (if both changed, only altitude is taken into account)
+	IF TG_OP = 'INSERT' THEN
+		IF NEW.altitude IS NULL THEN
+			NEW.altitude := ST_Z(NEW.geometry);
+		END IF;
+		IF ST_Z(NEW.geometry) IS NULL THEN
+			NEW.geometry := ST_MakePoint( ST_X(NEW.geometry), ST_Y(NEW.geometry), altitude );
+		END IF;
+	ELSIF TG_OP = 'UPDATE' THEN
+		IF NEW.altitue <> OLD.altitude THEN
```
Comment (tudorbarascu) [inline update/delta/delta_1.2.9_fix_valve_alt_sync.sql]: ``altitue`` typo

File: ordinary_data/valve/od_valve.sql [hunk 2]
```diff
@@ -154,6 +154,46 @@ CREATE TRIGGER tr_valve_altgeom_alt
...
+	-- altitude is prioritary on Z value of the geometry (if both changed, only altitude is taken into account)
+	IF TG_OP = 'INSERT' THEN
+		IF NEW.altitude IS NULL THEN
+			NEW.altitude := ST_Z(NEW.geometry);
+		END IF;
+		IF ST_Z(NEW.geometry) IS NULL THEN
+			NEW.geometry := ST_MakePoint( ST_X(NEW.geometry), ST_Y(NEW.geometry), altitude );
+		END IF;
+	ELSIF TG_OP = 'UPDATE' THEN
+		IF NEW.altitue <> OLD.altitude THEN
```
Comment (tudorbarascu) [inline ordinary_data/valve/od_valve.sql]: same typo here

**Comment by haubourg** [comment]
Closing 1.2.9 fixes it

**Pull request closed without merging**